repo2docker/contentproviders/zenodo.py

```diff
@@ -28,39 +29,59 @@ def _urlopen(self, req, headers=None):
 
         return urlopen(req)
 
-    def detect(self, doi, ref=None, extra_args=None):
-        """Trigger this provider for things that resolve to a Zenodo record"""
-        # To support Zenodo instances not hosted at zenodo.org we need to
-        # start maintaining a list of known DOI prefixes and their hostname.
-        # We should also change to returning a complete `record_url` that
-        # fetch() can use instead of constructing a URL there
-        doi = doi.lower()
-        # 10.5281 is the Zenodo DOI prefix
-        if doi.startswith("10.5281/"):
+    def _process_doi(self, doi):
```

Comment:
could we call it `doi2url` or `resolve_doi` or something? I had to read it a few times before I understood what it does and doesn't do (not helped in this case by the diff view :) ) ... I am a fan of names that tell you what the thing does.

Reply:
Yup!